src/Piral.Blazor.Tools/tasks/ManagePiletTask.cs

```diff
@@ -191,24 +191,24 @@ private int GetNpmVersion()
             return 0;
         }
 
-        private void CopyConfigurationFiles()
+        private void InitializeNpmConfiguration()
         {
-            var configurationFiles = new[] { ".npmrc" };
+            var configFileName = ".npmrc";
             var source = ConfigDir;
             var target = ProjectDir;
 
-            Log.LogMessage($"Copying config files from '{source}'...");
+            var existingConfigFile = Path.Combine(source, configFileName);
 
-            foreach (var configurationFile in configurationFiles)
+            if (File.Exists(existingConfigFile))
             {
-                var sourceFile = Path.Combine(source, configurationFile);
-
-                if (File.Exists(sourceFile))
-                {
-                    var targetFile = Path.Combine(target, configurationFile);
-                    Log.LogMessage($"Copying file '{configurationFile}' from '{sourceFile}'.");
-                    File.Copy(sourceFile, targetFile, true);
-                }
+                Log.LogWarning($"Ignoring NpmRegistry property! Copying file '{configFileName}' from '{existingConfigFile}'.");
+                var targetFile = Path.Combine(target, configFileName);
+                File.Copy(existingConfigFile, targetFile, true);
+            }
+            else
+            {
+                Log.LogMessage($"Creating '{configFileName}' with provided NpmRegistry property.");
+                File.AppendAllLines(Path.Combine(target, configFileName), [$"registry={NpmRegistry}", "always-auth=true"], Encoding.UTF8);
```

Comment:
We don't want to write this file if NpmRegistry was not set.

Reply:
NpmRegistry has default value of `https://registry.npmjs.org/` and we are adding default `.npmrc` like it was done previously by `pilet init` when `registry` was provided.. I thought we want to mimic the same?

Reply:
Yes we want to - but you don't check if there is something in NpmRegistry. What if the value is just an empty string?

Reply:
`<NpmRegistry></NpmRegistry>` would result error `The "ManagePiletTask" task was not given a value for the required parameter "NpmRegistry"`

So I'd say we are already covered with that and it's unnecessary to do empty check?

Reply:
Yeah but what if its just a single space character? I think this would still be valid input from perspective of MSBuild, but not be some input we'd like to take. Of course, this is just an edge case - so I wonder how practical that is.

Reply:
I tried single space character, it will result in same error

Reply:
Alright that means using the default `https://registry.npmjs.org/` there is no real way to detect if a value was just set explicitly or if the fallback is used.

In this case we should keep the logic you wrote here (just always write the file) and info log in case an the .npmrc file was found.